src/parser/parser.mly

```diff
@@ -251,12 +265,14 @@ typed_names:
   | xs=name+ COLON t=ty_term  { List.map (fun x -> (x, t)) xs }
 
 signature_clause:
-  | x=name COLON t=ty_term           { (x, None, t) }
-  | x=name AS y=name COLON t=ty_term { (x, Some y, t) }
+  | x=var_name COLON t=ty_term           { (x, None, t) }
+  | x=var_name AS y=name COLON t=ty_term { (x, Some y, t) }
 
 structure_clause :
-  | x=name EQ c=term                           { (x, None, c) }
-  | x=name AS y=name EQ c=term                 { (x, Some y, c) }
+  | x=var_name                                     { (x, None, None) }
```

Comment:
I am not sure I understand the syntax here. The user still has to mention all the fields, even the ones whose value will be determined by the sharing constraint?